nixos/modules/services/monitoring/uptime-kuma.nix

```diff
@@ -0,0 +1,110 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.uptime-kuma;
+  settingsToEnv = mapAttrsToList (name: value: "${name}=${value}");
+in
+{
+
+  options = {
+    services.uptime-kuma = {
+      enable = mkEnableOption "Uptime Kuma";
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/uptimekuma/";
+        description = "The directory where Uptime Kuma stores its data files.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "uptimekuma";
+        description = "User account under which Uptime Kuma runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "uptimekuma";
+        description = "Group under which Uptime Kuma runs.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        example = literalExpression "pkgs.uptime-kuma";
+        defaultText = "pkgs.uptime-kuma";
+        description = ''
+          Uptime Kuma package to use.
+        '';
+      };
+
+      settings = lib.mkOption {
+        type =
+          lib.types.submodule { freeformType = with lib.types; attrsOf str; };
+        default = { };
+        description = lib.mdDoc ''
+          Additional configuration for Uptime Kuma, see
+          <https://github.com/louislam/uptime-kuma/wiki/Environment-Variables">
+          for supported values.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.users = mkIf (cfg.user == "uptimekuma") {
+      uptimekuma = {
+        isSystemUser = true;
+        group = cfg.group;
+      };
+    };
+
+    users.groups = mkIf (cfg.group == "uptimekuma") { uptimekuma = { }; };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.uptime-kuma = {
+      description = "Uptime Kuma";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      # Database template has to be copied manually to set the right permissions
+      preStart = ''
+        if [ ! -f ${cfg.dataDir}/kuma.db ]; then
+          cp ${cfg.package}/db/kuma.db ${cfg.dataDir}/kuma.db
+          chmod 740 ${cfg.dataDir}/kuma.db
+        fi
+      '';
+      serviceConfig = {
+        Type = "simple";
+        Environment = [ "DATA_DIR=${cfg.dataDir}" ]
+          ++ settingsToEnv cfg.settings;
```

Comment:
Maybe you're looking for `systemd.services.uptimekuma.environment = cfg.settings;`?

Reply:
Yes! Thanks